VERSIONING.md

```diff
@@ -0,0 +1,50 @@
+= Rocket Chip Versioning Strategy
+
+== Version Naming
+
+* Rocket Chip uses a Calendar Versioning release strategy (https://calver.org/).
+* The format is YYYY.MM.DD.MINOR
```

Comment:
Just to be clear, the major release designation is the `YYYY.MM.DD` component. Minor releases will retain this designation (regardless of their release date), and simply bump the `MINOR` component. Perhaps a couple of examples would help clarify this.

Reply:
ah yes good point

Reply:
I just want to mention that this violates two different parts of CalVer, so we probably shouldn't call this CalVer:

- Major, Minor, and Micro are used to describe the ordinal position of the version component, with Minor specifically being used to refer to the second position.
- They discourage the use of 4-component versions, and there is nothing that matches the component that you are indicating with the "MINOR" designation above.

Reply:
"Major, Minor, and Micro are used to describe the ordinal position of the version component, with Minor specifically being used to refer to the second position.
"
Hm, that didn't match my read of the calver link posted above, which called out several examples of projects using various combinations of these things, but perhaps i should read it more closely to see if they still considered those "calver" or "calver-inspired"

Reply:
Ok, I see what you are saying. The only example on their site which violates the "dot-position of MAJOR/MINOR/MICRO defines what it is" is Teradata, which they explicitly call out a having a "useful combination of semvar and calver":
<img width="239" alt="Screen Shot 2020-04-04 at 2 59 32 AM" src="https://user-images.githubusercontent.com/17858596/78424176-6abca380-7620-11ea-9090-d6854163b7e5.png">

Reply:
There is also this one that seems to use an `_` to get around the "MICRO is in the wrong spot" but they don't give any further explanation on whether that is encouraged/discouraged:
<img width="856" alt="Screen Shot 2020-04-04 at 3 03 47 AM" src="https://user-images.githubusercontent.com/17858596/78424269-2aa9f080-7621-11ea-9a7f-b31be73f143d.png">

Reply:
@richardxia do you think that we "shouldn't call this calver" or should change it until it matches calver (which presumably is a good thing)? For example I can't tell if using something like `YYYY_0M_0D.MINOR` would address your points since there aren't a lot of "underscore" examples.

Reply:
I think we should pop up a level to discuss what kinds of compatibility guarantees we want to provide and what our branching strategy and bugfix policy will be. Despite using dates in the format string, you seem to want to provide some notion of semantic versioning by adopting a definition of "minor" that is equivalent to SemVer's. If this is important to us, then we shouldn't obfuscate it using CalVer.

You can see that in the examples on the calver.org website, besides the `MICRO` version components, the projects their either fall into the category of "every non-micro release is guaranteed to break compatibility and we are not trying to address that at all" (e.g. Ubuntu) or "my library is basically permanently backwards compatible, but I am dependent on stuff that changes over time, like time zones or SSL certificate bundles". Most of the libraries on that page which have a `MICRO` version only really use it for bugfixes or security patches.

Note that you _are_ allowed to use SemVer while trying to stick to a date-based release schedule; you can fully adopt SemVer's versioning schemes, and instead allowing changes to dictate SemVer version bumps, you can let your release schedule dictate what kinds of changes are allowed to be made.